shell/platform/fuchsia/flutter/engine.cc

```diff
@@ -861,24 +861,24 @@ void Engine::WarmupSkps(
     std::shared_ptr<flutter::AssetManager> asset_manager,
     std::optional<const std::vector<std::string>> skp_names,
     std::optional<std::function<void(uint32_t)>> completion_callback) {
-  // We use a raw pointer here because we want to keep this alive until all gpu
-  // work is done and the callbacks skia takes for this are function pointers
-  // so we are unable to use a lambda that captures the smart pointer.
-  SurfaceProducerSurface* skp_warmup_surface =
-      surface_producer->ProduceOffscreenSurface(size).release();
-  if (!skp_warmup_surface) {
-    FML_LOG(ERROR) << "Failed to create offscreen warmup surface";
-    // Tell client that zero shaders were warmed up because warmup failed.
-    if (completion_callback.has_value() && completion_callback.value()) {
-      completion_callback.value()(0);
-    }
-    return;
-  }
+  // We use this bizzare raw pointer to a smart pointer thing here because we
+  // want to keep the surface alive until all gpu work is done and the
+  // callbacks skia takes for this are function pointers so we are unable to
+  // use a lambda that captures the smart pointer. We need two levels of
+  // indirection because it needs to be the same across all invocations of the
+  // raster task lambda from a single invocation of WarmupSkps, but be
+  // different across different invocations of WarmupSkps (so we cant
+  // statically initialialize it in the lambda itself). Basically the result
+  // of a mashup of wierd call dynamics, multithreading, and lifecycle
+  // management with C style Skia callbacks.
+  std::unique_ptr<SurfaceProducerSurface>* skp_warmup_surface =
```

Comment:
Is it okay that we're doing `delete static_cast<SurfaceProducerSurface*>(skp_warmup_surface);` on this below as though it were a `SurfaceProducerSurface*` instead of a `unique_ptr<SurfaceProducerSurface*>`?

Reply:
NVM, I see that `skp_warmup_surface` is redefined within that scope.

Reply:
no you're right thats very bad and will cause crashes. Good catch. Looks like the tests werent catching this because they tear down the context before it completes so the callback never runs. I'll fix this and plumb the tests to wait the for completion properly so theyll catch things like this.